File: AstraNews/app/src/main/java/com/example/astranews/adapter/RecyclerAdapter.java
package com.example.astranews.adapter;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import com.example.astranews.R;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.astranews.dataclasses.JsonClass;
import com.squareup.picasso.Picasso;


public class RecyclerAdapter extends RecyclerView.Adapter<NewsViewHolder>{

    JsonClass[] items;
    @NonNull
    @Override
    public NewsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new NewsViewHolder(LayoutInflater.from(parent.getContext()).inflate(com.example.astranews.R.layout.news_item, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull NewsViewHolder holder, int position) {
        JsonClass current = items[position];
        holder.title.setText(current.newsTitle);
        holder.summary.setText(current.newsSummary);
        Picasso.get().load(current.imageUrl).into(holder.imageView);
    }

    @Override
    public int getItemCount() {
        return items.length;
    }
}

class NewsViewHolder extends RecyclerView.ViewHolder{
    public ImageView imageView;
    public TextView title;
    public TextView summary;
    View view;

    public NewsViewHolder(@NonNull View itemView) {
        super(itemView);
        imageView = (ImageView) itemView.findViewById(com.example.astranews.R.id.NewsImage);
        title = (TextView) itemView.findViewById(com.example.astranews.R.id.NewsTitle);
        summary = (TextView) itemView.findViewById(com.example.astranews.R.id.NewsSummary);
        view = itemView;
    }
}

File: Photos/app/src/main/java/com/example/photos/adapters/RecyclerAdapter.kt
package com.example.photos.adapters

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.photos.dataclasses.UrlClass
import com.example.photos.R
import com.squareup.picasso.Picasso

//recycler adapter for displaying list of images
class RecyclerAdapter : RecyclerView.Adapter<ViewHolder>() {

    //list that stores the items
    var items: ArrayList<UrlClass> = ArrayList()

    //is called when a new ViewHolder of given type is required to display an item
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        //inflate the view for an individual item
        val view = LayoutInflater.from(parent.context).inflate(R.layout.image_view, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        //returns the total number of items held by adapter
        return items.size
    }

    //is called to display the data at specified position
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        //get data for current position
        val current = items[position]

        //bind the data to the view
        holder.title.text = current.alt_description
        holder.author.text = current.user?.name
        Picasso.get().load(current.urls?.thumb).into(holder.image)
    }

}

//View holder to hold the view of an individual item in the recycler view
class ViewHolder(itemView: View): RecyclerView.ViewHolder(itemView){
    val image: ImageView = itemView.findViewById(R.id.imageView1)
    val title: TextView = itemView.findViewById(R.id.title1)
    val author: TextView = itemView.findViewById(R.id.author1)
}
